src/main/java/com/hermesworld/ais/galapagos/kafka/impl/ConnectedKafkaCluster.java

```diff
@@ -137,14 +138,12 @@ public String getId() {
         ConfigResource res = new ConfigResource(Type.TOPIC, topicName);
         Config config = new Config(configs);
 
-        return toCompletableFuture(
-                this.adminClient
-                        .createTopics(
-                                Collections.singleton(new NewTopic(topicName, topicCreateParams.getNumberOfPartitions(),
-                                        (short) topicCreateParams.getReplicationFactor())))
-                        .all()).thenCompose(
-                                o -> toCompletableFuture(
-                                        adminClient.alterConfigs(Collections.singletonMap(res, config)).all()));
+        // TODO numPartitions should be configurable for Galapagos metadata topics
+        NewTopic newTopic = new NewTopic(topicName, 3, (short) topicCreateParams.getReplicationFactor());
+        newTopic = newTopic.configs(Map.of(TopicConfig.CLEANUP_POLICY_CONFIG, TopicConfig.CLEANUP_POLICY_COMPACT));
+
+        return toCompletableFuture(this.adminClient.createTopics(Collections.singleton(newTopic)).all()).thenCompose(
+                o -> toCompletableFuture(adminClient.alterConfigs(Collections.singletonMap(res, config)).all()));
```

Comment:
No, this is the errorneous code from before. Should still cause the issue on Confluent Cloud. Well, could be fine because now it is set in the first call, so this is a no-op.
But just remove the `.thenCompose` and everything from there, and you are fine here. (Configs are in `newTopic`)